presto-parser/src/main/java/com/facebook/presto/sql/tree/CreateType.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public class CreateType
+        extends Statement
+{
+    private final QualifiedName typeName;
+    private final Optional<String> distinctType;
+    private final List<String> parameterNames;
+    private final List<String> parameterTypes;
+
+    public CreateType(QualifiedName typeName, Optional<String> distinctType, List<String> parameterNames, List<String> parameterTypes)
+    {
+        super(Optional.empty());
+
+        checkArgument(parameterNames.size() == parameterTypes.size(), "Expected one type for each parameter name");
+        checkArgument(distinctType.isPresent() || !parameterNames.isEmpty(), "Type can be either a distinct type or parameterized");
+
+        this.typeName = typeName;
+        this.distinctType = distinctType;
+        this.parameterNames = parameterNames;
+        this.parameterTypes = parameterTypes;
+
+        // Check for duplicate fields
+        HashSet<String> parameterNamesSet = new HashSet();
+        for (int i = 0; i < parameterNames.size(); i++) {
+            String parameterName = parameterNames.get(i);
+            String parameterType = parameterTypes.get(i);
+            checkArgument(!parameterNamesSet.contains(parameterName), format("Type %s has multiple parameters for %s", typeName, parameterType));
```

Comment:
The error message is not very clear, it would say something like `Type foo has multiple parameters for bigint`. It might be more meaningful to say something like "Duplicated member definition for [parameter_name] in type [type_name]", you can optionally list all the types but that's less important as the error is mostly in the name rather than the type.

Also, this should be a syntax error, not sure whether there's anything catching this and convert it to a syntax error, just double check.

Another nits, we generally prefer using stream API when position is not important (if you don't use type in the error message, that would be the case).